src/server/connectors/address-lookup/organisationRemovalFromLookup.js

```diff
@@ -0,0 +1,75 @@
+const { logEmitter } = require("../../services/logging.service");
+/**
+ * Removes organization from address lookup
+ *
+ * @param {array} addressList A list of addresses
+ *
+ * @returns {array} A list of addresses without organization
+ */
+const organisationRemovalFromLookup = (addressList) => {
+  logEmitter.emit(
+    "functionCall",
+    "organisationRemovalFromLookup",
+    "organisationRemovalFromLookup"
+  );
+  if (addressList.length === 0) {
+    return addressList;
+  }
+  const addressListWithoutOrganisation = addressList.map((a) => {
+    var address = a;
+    if (address.hasOwnProperty("organisation")) {
+      // removes organisation from summaryline
+      if (
+        isString(address.organisation) &&
+        address.summaryline.startsWith(address.organisation + ", ")
+      ) {
+        address.summaryline = address.summaryline.substring(
+          address.organisation.length + 2
+        );
+      }
+      // removes organisation key
+      delete address.organisation;
+    }
+    logEmitter.emit(
+      "functionSuccess",
+      "organisationRemovalFromLookup",
+      "organisationRemovalFromLookup"
+    );
+    return address;
+  });
+  return removeDuplicateObjectFromArray(
+    addressListWithoutOrganisation,
+    "summaryline"
+  );
+};
+
+/**
+ * Removes duplicate objects from array by key
+ *
+ * @param {array} array of objects
+ * @param {key} object key to be compared
+ *
+ * @returns {array} array of objects without duplicates
+ */
+const removeDuplicateObjectFromArray = (array, key) => {
```

Comment:
can replace function body with: `return [...new Set(arr)];` and remove key parameter. Probably needs a comment to make it clear what it's doing as perhaps not 100% as intuitive as your approach but more concise and not overly-complex

Reply:
Your approach seems good and is concise